clients/cpp-console/src/common/mainAudio.cpp

```diff
@@ -66,26 +84,54 @@ int main(int argc, char** argv)
     string configFilePath = argv[1];
     string s;
     const char * device = "default";
-    bool keywordListeningEnabled = false;
+    KeywordActivationState keywordActivationState = KeywordActivationState::Undefined;
     bool volumeOn = false;
     
     IAudioPlayer* player;
     shared_ptr<AgentConfiguration> agentConfig;
     shared_ptr<DialogServiceConnector> dialogServiceConnector;
     
-    auto startKwsIfApplicable = [&]()
+    auto StartKws = [&]()
     {
-        log_t("startKWS called");
-        if (keywordListeningEnabled)
+        log_t("startKwsIfApplicable called");
+
+        if (KeywordActivationState::Paused == keywordActivationState)
         {
             auto modelPath = agentConfig->KeywordModel();
             log_t("Initializing keyword recognition with: ", modelPath);
             auto model = KeywordRecognitionModel::FromFile(modelPath);
             auto _ = dialogServiceConnector->StartKeywordRecognitionAsync(model);
+            keywordActivationState = KeywordActivationState::Listening;
             log_t("KWS initialized");
         }
-        else {
-            log_t("no model file specified. Cannot start keyword listening");
+    };
+
+    auto PauseKws = [&]()
+    {
+        log_t("PauseKws called");
```

Comment:
on these logs should we print out what the state was when being called?

Reply:
Added printout of the KWS state on entry and exit of the functions. Thanks!